### DISCUSSION TITLE
Can't get it to work with KMK - tried with both circuitpython 7.x and 8.x

### DISCUSSION THREAD

**Discussion by raghur** [manna-harbour/xmk#10] [Q&A]
Hello,

I can't seem to get this going either with circuit python 7.x or 8.x. MCU - RPI Pico
Running 
```
./src/xmk -k /dev/input/by-id/usb-SIGMACHIP_USB_Keyboard-event-kbd -m ./src/maps/minidox/60_ansi -s > /dev/ttyACM1
```
with the keyboard event and the device ttyACM1 device just produces a constant stream of CRs on the console

Steps I followed:
1. Installed circuit python 7.3.3 uf2 file
2. git cloned kmk fork - `origin  https://github.com/manna-harbour/kmk_firmware.git (fetch)`
3. added the KMK repo as `upstream` - `upstream        https://github.com/KMKfw/kmk_firmware (fetch)`
4. `git fetch upstream` & `git merge upstream/master`
5. Copied `kmk` folder to the root of the pico
6. copied `boards/xmk` to the root of the pico
 
![image](https://github.com/manna-harbour/xmk/assets/516267/09b5edfe-901b-4b37-93d8-b48c856679d5)

Other things I had to do
1. I'm on Manjaro - so had trouble with the perms for hte ttyACMx device - so did a `sudo chmod a+rw` to give perms 
![image](https://github.com/manna-harbour/xmk/assets/516267/44d5b382-8123-46a3-9751-56ac1fae06aa)

**Comment by manna-harbour** [comment]
Try without the redirection into the tty, to check the output of xmk. E.g. `./src/xmk -k /dev/input/by-id/usb-SIGMACHIP_USB_Keyboard-event-kbd -m ./src/maps/minidox/60_ansi -s`.

Also try directly from https://github.com/manna-harbour/kmk_firmware i.e. without merging upstream.

**Reply by raghur** [reply]
Ok - 
1. tried directly from [mann](https://github.com/manna-harbour/kmk_firmware) without merging from upstream
2. removed redirection as suggested

Results in again the console just running away with CR - pressing keys in between echoes key press and key release events.

Should there be a constant stream of <cr> keys to the console?
```
key press 29


key release 29



key press 29



key release 29



key press 29


key release 29




key press 29



key release 29
```
